### PR TITLE
Avoid values out of bound, and simplify logic in PowerConverters.DCAC.Control.SVPWM

### LINKED ISSUE #3887: Inadequate handling of geometric corner case in PowerConverters.DCAC.Control.SVPWM
During simulation of `Modelica.Electrical.PowerConverters.Examples.DCAC.PolyphaseTwoLevel.ThreePhaseTwoLevel_PWM` I encounter the following problem under very specific conditions, leading to `pwm.svPWM.ka` = 6, which then causes array indexing out of bounds.

The problem happens when the `when`-statement in `Modelica.Electrical.PowerConverters.DCAC.Control.SVPWM` is triggered and one has (happening very close to time 0.5):
- `pwm.svPWM.u[1]`: 1.4142135623730951
- `pwm.svPWM.u[2]`: -3.4638242249419736e-16

Based on these numbers I then obtain:
- `pwm.svPWM.phiRef`: -2.4492935982947064e-16
- `pwm.svPWM.phiPos`: 6.2831853071795862

The `pwm.svPWM.phiPos` is equal to `2 * pi` since `pwm.svPWM.phiRef` is too small to make a difference.

For `m` = 3 we have:
- `pi / m`: 1.0471975511965976
- `pwm.svPWM.phiPos / (pi / m)`: 6 (formatted with `%.17g`)

After all, 6 should be the expected result once you have landed in `pwm.svPWM.phiPos` = `2 * pi`, and it should be clear that the model needs to be adjusted somehow to not end up in this geometric corner case.

### PR DESCRIPTION
**Pull request by HansOlsson**
Closes #3887

### CODE REVIEW

File: Modelica/Electrical/PowerConverters/DCAC/Control/SVPWM.mo [hunk 1]
```diff
@@ -33,12 +33,14 @@ algorithm
     //Limited relative reference signal
     uRef:=min(sqrt(u[1]^2 + u[2]^2)/(2/3*uMax), cos(pi/6));
     //Determine angle of reference signal within (-pi, +pi]
-    phiRef:=if noEvent(uRef < small) then 0 else atan2(u[2], u[1]);
+    phiRef:=if uRef < small then 0 else atan2(u[2], u[1]);
     //Shift angle to [0, 2*pi)
-    phiPos:=max(phiRef + (if phiRef < -eps then 2*pi else 0), 0);
+    // Note that phiRef is likely to be very close to zero, so the extra logic ensures that
+    // we do not get values at 2*pi since that will look odd.
+    phiPos:=max(0, phiRef + (if phiRef < -2*pi*Modelica.Constants.eps then 2*pi else 0));
```
Comment (henrikt-ma) [inline Modelica/Electrical/PowerConverters/DCAC/Control/SVPWM.mo:40]: I can see the logic of choosing the factor `2 ^ pi` in front of `eps`, but it is probably not obvious to someone who isn't familiar with the peculiar definition of `eps` in the MSL.  Might be worth explaining what's going on in a comment…

**Review by henrikt-ma — APPROVED** [review]
Looks like a good fix.

For what it's worth, it also makes the `ThreePhaseTwoLevel_PWM` example simulate fine with System Modeler.

File: Modelica/Electrical/PowerConverters/DCAC/Control/SVPWM.mo (hunk 1 above)
Comment (beutlich) [reply Modelica/Electrical/PowerConverters/DCAC/Control/SVPWM.mo:40]: Could Modelica.Math.wrapAngle be utilized here?

Comment (AHaumer) [reply Modelica/Electrical/PowerConverters/DCAC/Control/SVPWM.mo:40]: nope, tested: results in "Index out of bounds".

**Review by AHaumer — APPROVED** [review]
fine with me, thanks @HansOlsson for implementing this solution. 
Reading a text book, this algorithm is pretty easy to understand. 
Fine avoiding numerical issues with as many tools as possible.

**Pull request merged**